File: 0009-palindrome-number/0009-palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        if(x<0){
            return false;
        }
        int dup=x;
        long long rev=0;
        while(x>0){
            rev=rev*10+x%10;
            x/=10;
        }
        if(dup==rev){
            return true;
        }
        return false;
    }
};

File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        if(x<0)return false;
        int p=0;
        int dup=x;
        while(x!=0){
            p=p*10+x%10;
            x/=10;
        }
        if(p==dup)return true;
        return false;
    }
}